disperser/apiserver/server.go

```diff
@@ -457,6 +457,11 @@ func (s *DispersalServer) checkRateLimitsAndAddRatesToHeader(ctx context.Context
 		// Update the quorum rate
 		blob.RequestHeader.SecurityParams[i].QuorumRate = accountRates.Throughput
 
+		// Update AccountID to accountKey.
+		// This will use the origin as the account key if the user does not provide
+		// an authenticated address.
+		blob.RequestHeader.BlobAuthHeader.AccountID = accountKey
```

Comment:
check if `blob.RequestHeader.BlobAuthHeader` is nil?

Reply:
validateRequestAndGetBlob already makes sure that the blob is defined properly.